### PR TITLE
[mercadopago] Fix types for `SimpleAddress` interface

### PR DESCRIPTION
**Pull request by MysteryBlokHed**
Related discussion: #62652

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test mercadopago`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://www.mercadopago.com.mx/developers/en/reference/payments/_payments/post>
  The poster in #62652 has also said that the types are optional from their own experience
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## mercadopago ([<kbd>unpkg</kbd>](https://unpkg.com/browse/mercadopago@latest/))
was missing the following properties:
1. The declaration doesn't match the JavaScript module 'mercadopago'. Reason:
The JavaScript module can be called or constructed, but the declaration module cannot.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against e98f8eade80a500b692bb585169c385e67b23ded
</p>

File: types/mercadopago/shared/address.d.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 export interface SimpleAddress {
-  zip_code: string;
-  street_name: string;
-  street_number: string;
+  zip_code?: string | undefined;
+  street_name?: string | undefined;
+  street_number?: number | undefined;
```
Comment (danieldspx) [inline types/mercadopago/shared/address.d.ts:4]:
```suggestion
  zip_code?: string;
  street_name?: string;
  street_number?: number;
```

**Review by danieldspx — CHANGES_REQUESTED** [review]

File: types/mercadopago/shared/address.d.ts (hunk 1 above)
Comment (MysteryBlokHed) [reply types/mercadopago/shared/address.d.ts:4]: If a user is using TypeScript's `exactOptionalPropertyTypes`, then removing the `| undefined` will prevent them explicitly pass `undefined` for an optional value

Comment (danieldspx) [reply types/mercadopago/shared/address.d.ts:4]: True

**Review by danieldspx — APPROVED** [review]

**Pull request merged**